### PR TITLE
Adding What is Pulumi details to core concepts page

### PR DESCRIPTION
**Pull request by interurban**
## Description
adding more context about what is pulumi and lang/sdk coverage, addressing SEO gaps

### CODE REVIEW

**Review by cnunciato — COMMENTED** [review]
Added a commit with a few edits, and fixed the example, which was broken in several ways. 😅

File: themes/default/content/docs/concepts/_index.md [hunk 1]
```diff
@@ -15,262 +14,62 @@ aliases:
 - /docs/intro/concepts/
 ---
 
-Pulumi is a modern [infrastructure as code](/what-is/what-is-infrastructure-as-code/) platform. It leverages existing programming languages---TypeScript, JavaScript, Python, Go, .NET, Java, and markup languages like YAML---and their native ecosystem to interact with cloud resources through the Pulumi SDK. A downloadable CLI, runtime, libraries, and a hosted service work together to deliver a robust way of provisioning, updating, and managing cloud infrastructure.
+Pulumi is a modern [infrastructure as code](/what-is/what-is-infrastructure-as-code/) platform. It leverages existing programming languages---TypeScript, JavaScript, Python, Go, .NET, Java, and markup languages like YAML---and their native ecosystems to interact with cloud resources through the Pulumi SDK. A [downloadable CLI](/docs/install/), runtime, libraries, and a hosted service work together to deliver a robust platform for of provisioning, updating, and managing cloud infrastructure.
```
Comment (ericrudder) [inline themes/default/content/docs/concepts/_index.md:17]:
```suggestion
Pulumi is a modern [infrastructure as code](/what-is/what-is-infrastructure-as-code/) platform. It leverages existing programming languages---TypeScript, JavaScript, Python, Go, .NET, Java, and markup languages like YAML---and their native ecosystems to interact with cloud resources. A [downloadable CLI](/docs/install/), runtime, libraries, and a hosted service work together to deliver a robust platform for provisioning, updating, and managing cloud infrastructure.
```

File: themes/default/content/docs/concepts/_index.md [hunk 2]
```diff
@@ -15,262 +14,62 @@ aliases:
...
-});
-
-export const publicIp = server.publicIp;
-export const publicDns = server.publicDns;
-```
+Pulumi is an [infrastructure as code](/what-is/what-is-infrastructure-as-code/) platform that allows you to use familiar programming languages and tools to build, deploy, and manage cloud infrastructure.
 
-{{% /choosable %}}
-{{% choosable language typescript %}}
+Pulumi is free, [open source](https://github.com/pulumi/pulumi), and optionally pairs with the [Pulumi Cloud](https://www.pulumi.com/docs/pulumi-cloud/) to make managing infrastructure secure, reliable, and hassle-free.
```
Comment (cnunciato) [inline themes/default/content/docs/concepts/_index.md:27]:
```suggestion
Pulumi is free, [open source](https://github.com/pulumi/pulumi), and optionally pairs with [Pulumi Cloud](https://www.pulumi.com/docs/pulumi-cloud/) to make managing infrastructure secure, reliable, and hassle-free.
```

File: themes/default/content/docs/concepts/_index.md [hunk 3]
```diff
@@ -15,262 +14,62 @@ aliases:
...
-    ami='ami-6869aa05',
-    instance_type='t2.micro',
-    vpc_security_group_ids=[group.name] # reference the security group resource above
-)
+- **Command-Line interface (CLI)**: Pulumi is controlled primarily using the command line interface [(CLI)](https://www.pulumi.com/docs/cli/). It works in conjunction with the [Pulumi Cloud](https://www.pulumi.com/docs/pulumi-cloud/) to deploy changes to your cloud apps and infrastructure. It keeps a history of who updated what in your team and when. This CLI has been designed for great inner loop productivity, in addition to continuous integration and delivery scenarios.
 
-pulumi.export('public_ip', server.public_ip)
-pulumi.export('public_dns', server.public_dns)
-```
+- **Deployment engine** The deployment engine is responsible for computing the set of operations needed to drive the current state of your infrastructure into the desired state expressed by your program.
```
Comment (cnunciato) [inline themes/default/content/docs/concepts/_index.md:52]:
```suggestion
- **Deployment engine**: The deployment engine is responsible for computing the set of operations needed to drive the current state of your infrastructure into the desired state expressed by your program.
```

**Review by cnunciato — APPROVED** [review]
Looks good! Only a couple of super-teeny things I happened to notice.

File: themes/default/content/docs/concepts/_index.md [hunk 4]
```diff
@@ -15,262 +14,62 @@ aliases:
...
-        { 'protocol': 'tcp', 'from_port': 80, 'to_port': 80, 'cidr_blocks': ['0.0.0.0/0'] }
-    ])
+- **Software development kit (SDK)**: Pulumi Software Development Kit (SDK) provides bindings for each type of resource that the provider can manage. This provides the necessary tools and libraries for defining and managing cloud resources on any cloud and with any provider.
 
-server = aws.ec2.Instance('web-server',
-    ami='ami-6869aa05',
-    instance_type='t2.micro',
-    vpc_security_group_ids=[group.name] # reference the security group resource above
-)
+- **Command-Line interface (CLI)**: Pulumi is controlled primarily using the command line interface [(CLI)](https://www.pulumi.com/docs/cli/). It works in conjunction with the [Pulumi Cloud](https://www.pulumi.com/docs/pulumi-cloud/) to deploy changes to your cloud apps and infrastructure. It keeps a history of who updated what in your team and when. This CLI has been designed for great inner loop productivity, in addition to continuous integration and delivery scenarios.
```
Comment (scottslowe) [inline themes/default/content/docs/concepts/_index.md:50]:
```suggestion
- **Command-Line interface (CLI)**: Pulumi is controlled primarily using the command line interface [(CLI)](https://www.pulumi.com/docs/cli/). It works in conjunction with [Pulumi Cloud](https://www.pulumi.com/docs/pulumi-cloud/) to deploy changes to your cloud apps and infrastructure. It keeps a history of who updated what in your team and when. This CLI has been designed for great inner loop productivity, in addition to continuous integration and delivery scenarios.
```

File: themes/default/content/docs/concepts/_index.md [hunk 5]
```diff
@@ -15,262 +14,62 @@ aliases:
...
-import com.pulumi.Pulumi;
-import com.pulumi.aws.ec2.Instance;
-import com.pulumi.aws.ec2.InstanceArgs;
-import com.pulumi.aws.ec2.SecurityGroup;
-import com.pulumi.aws.ec2.SecurityGroupArgs;
-import com.pulumi.aws.ec2.inputs.SecurityGroupIngressArgs;
+This diagram illustrates the structure and major components of Pulumi.
 
-import java.util.List;
+![Pulumi programming model diagram.](/images/docs/pulumi-programming-model-diagram.svg)
```
Comment (scottslowe) [inline themes/default/content/docs/concepts/_index.md:56]: It would be nice to have a graphic designer/artist give this diagram a refresh (my 2 cents).

**Review by scottslowe — APPROVED** [review]
One suggestion and one comment, but nothing needs to block merging the PR. I do agree with Christian's suggestions/comments as well.

File: themes/default/content/docs/concepts/_index.md (hunk 5 above)
Comment (interurban) [reply themes/default/content/docs/concepts/_index.md:56]: good call; well add that in when its complete.

**Pull request merged**